src/Http/Http.Extensions/src/HttpResponseJsonExtensions.cs

```diff
@@ -89,13 +90,23 @@ public static partial class HttpResponseJsonExtensions
 
         response.ContentType = contentType ?? JsonConstants.JsonContentTypeWithCharset;
 
+        var startTask = Task.CompletedTask;
+        if (!response.HasStarted)
+        {
+            // Flush headers before starting Json serialization. This avoids an extra layer of buffering before the first flush.
+            startTask = response.StartAsync(cancellationToken);
```

Comment:
It'd be nice to get some HTTP/2 numbers too. I don't think this optimization helps in that case, but it probably doesn't hurt either. @sebastienros Do we have any HTTP/2 scenarios that use `WriteAsJsonAsync`?